packages/backends/postgres/src/postgres-backend.ts

```diff
@@ -2,29 +2,55 @@ import { SQLBackend } from "@sidequest/backend";
 import createKnex, { Knex } from "knex";
 import path from "path";
 
+/**
+ * Configuration for PostgreSQL backend that supports full Knex configuration options.
+ */
+export type PostgresBackendConfig = Pick<Knex.Config, "pool" | "connection">;
```

Comment:
The goal here is to limit the options we can pass to Knex. I didnt want to allow overriding _everything_, just the pool and connection options (which you already partially allow via the `Knex.ConnectionConfig` typing).